File: auto-generated/ruby/lib/swagger_client/models/address.rb
=begin
#BitMEX API

### REST API for the BitMEX Trading Platform  _If you are building automated tools, please subscribe to the_ _[BitMEX API RSS Feed](https://blog.bitmex.com/api_announcement/feed/) for changes. The feed will be updated_ _regularly and is the most reliable way to get downtime and update announcements._  [View Changelog](/app/apiChangelog)  -  #### Getting Started  Base URI: [https://www.bitmex.com/api/v1](/api/v1)  ##### Fetching Data  All REST endpoints are documented below. You can try out any query right from this interface.  Most table queries accept `count`, `start`, and `reverse` params. Set `reverse=true` to get rows newest-first.  Additional documentation regarding filters, timestamps, and authentication is available in [the main API documentation](/app/restAPI).  _All_ table data is available via the [Websocket](/app/wsAPI). We highly recommend using the socket if you want to have the quickest possible data without being subject to ratelimits.  ##### Return Types  By default, all data is returned as JSON. Send `?_format=csv` to get CSV data or `?_format=xml` to get XML data.  ##### Trade Data Queries  _This is only a small subset of what is available, to get you started._  Fill in the parameters and click the `Try it out!` button to try any of these queries.  - [Pricing Data](#!/Quote/Quote_get)  - [Trade Data](#!/Trade/Trade_get)  - [OrderBook Data](#!/OrderBook/OrderBook_getL2)  - [Settlement Data](#!/Settlement/Settlement_get)  - [Exchange Statistics](#!/Stats/Stats_history)  Every function of the BitMEX.com platform is exposed here and documented. Many more functions are available.  ##### Swagger Specification  [⇩ Download Swagger JSON](swagger.json)  -  ## All API Endpoints  Click to expand a section. 

OpenAPI spec version: 1.2.0
Contact: support@bitmex.com
Generated by: https://github.com/swagger-api/swagger-codegen.git
Swagger Codegen version: 2.4.42-SNAPSHOT

=end

require 'date'

module SwaggerClient
  class Address
    attr_accessor :id

    attr_accessor :currency

    attr_accessor :created

    attr_accessor :user_id

    attr_accessor :address

    attr_accessor :name

    attr_accessor :note

    attr_accessor :skip_confirm

    attr_accessor :skip_confirm_verified

    attr_accessor :skip2_fa

    attr_accessor :skip2_fa_verified

    attr_accessor :network

    attr_accessor :memo

    # Attribute mapping from ruby-style variable name to JSON key.
    def self.attribute_map
      {
        :'id' => :'id',
        :'currency' => :'currency',
        :'created' => :'created',
        :'user_id' => :'userId',
        :'address' => :'address',
        :'name' => :'name',
        :'note' => :'note',
        :'skip_confirm' => :'skipConfirm',
        :'skip_confirm_verified' => :'skipConfirmVerified',
        :'skip2_fa' => :'skip2FA',
        :'skip2_fa_verified' => :'skip2FAVerified',
        :'network' => :'network',
        :'memo' => :'memo'
      }
    end

    # Attribute type mapping.
    def self.swagger_types
      {
        :'id' => :'Float',
        :'currency' => :'String',
        :'created' => :'DateTime',
        :'user_id' => :'Float',
        :'address' => :'String',
        :'name' => :'String',
        :'note' => :'String',
        :'skip_confirm' => :'BOOLEAN',
        :'skip_confirm_verified' => :'BOOLEAN',
        :'skip2_fa' => :'BOOLEAN',
        :'skip2_fa_verified' => :'BOOLEAN',
        :'network' => :'String',
        :'memo' => :'String'
      }
    end

    # Initializes the object
    # @param [Hash] attributes Model attributes in the form of hash
    def initialize(attributes = {})
      return unless attributes.is_a?(Hash)

      # convert string to symbol for hash key
      attributes = attributes.each_with_object({}) { |(k, v), h| h[k.to_sym] = v }

      if attributes.has_key?(:'id')
        self.id = attributes[:'id']
      end

      if attributes.has_key?(:'currency')
        self.currency = attributes[:'currency']
      end

      if attributes.has_key?(:'created')
        self.created = attributes[:'created']
      end

      if attributes.has_key?(:'userId')
        self.user_id = attributes[:'userId']
      end

      if attributes.has_key?(:'address')
        self.address = attributes[:'address']
      end

      if attributes.has_key?(:'name')
        self.name = attributes[:'name']
      end

      if attributes.has_key?(:'note')
        self.note = attributes[:'note']
      end

      if attributes.has_key?(:'skipConfirm')
        self.skip_confirm = attributes[:'skipConfirm']
      end

      if attributes.has_key?(:'skipConfirmVerified')
        self.skip_confirm_verified = attributes[:'skipConfirmVerified']
      end

      if attributes.has_key?(:'skip2FA')
        self.skip2_fa = attributes[:'skip2FA']
      end

      if attributes.has_key?(:'skip2FAVerified')
        self.skip2_fa_verified = attributes[:'skip2FAVerified']
      end

      if attributes.has_key?(:'network')
        self.network = attributes[:'network']
      end

      if attributes.has_key?(:'memo')
        self.memo = attributes[:'memo']
      end
    end

    # Show invalid properties with the reasons. Usually used together with valid?
    # @return Array for valid properties with the reasons
    def list_invalid_properties
      invalid_properties = Array.new
      if @address.nil?
        invalid_properties.push('invalid value for "address", address cannot be nil.')
      end

      if @name.nil?
        invalid_properties.push('invalid value for "name", name cannot be nil.')
      end

      if @network.nil?
        invalid_properties.push('invalid value for "network", network cannot be nil.')
      end

      invalid_properties
    end

    # Check to see if the all the properties in the model are valid
    # @return true if the model is valid
    def valid?
      return false if @address.nil?
      return false if @name.nil?
      return false if @network.nil?
      true
    end

    # Checks equality by comparing each attribute.
    # @param [Object] Object to be compared
    def ==(o)
      return true if self.equal?(o)
      self.class == o.class &&
          id == o.id &&
          currency == o.currency &&
          created == o.created &&
          user_id == o.user_id &&
          address == o.address &&
          name == o.name &&
          note == o.note &&
          skip_confirm == o.skip_confirm &&
          skip_confirm_verified == o.skip_confirm_verified &&
          skip2_fa == o.skip2_fa &&
          skip2_fa_verified == o.skip2_fa_verified &&
          network == o.network &&
          memo == o.memo
    end

    # @see the `==` method
    # @param [Object] Object to be compared
    def eql?(o)
      self == o
    end

    # Calculates hash code according to all attributes.
    # @return [Fixnum] Hash code
    def hash
      [id, currency, created, user_id, address, name, note, skip_confirm, skip_confirm_verified, skip2_fa, skip2_fa_verified, network, memo].hash
    end

    # Builds the object from hash
    # @param [Hash] attributes Model attributes in the form of hash
    # @return [Object] Returns the model itself
    def build_from_hash(attributes)
      return nil unless attributes.is_a?(Hash)
      self.class.swagger_types.each_pair do |key, type|
        if type =~ /\AArray<(.*)>/i
          # check to ensure the input is an array given that the attribute
          # is documented as an array but the input is not
          if attributes[self.class.attribute_map[key]].is_a?(Array)
            self.send("#{key}=", attributes[self.class.attribute_map[key]].map { |v| _deserialize($1, v) })
          end
        elsif !attributes[self.class.attribute_map[key]].nil?
          self.send("#{key}=", _deserialize(type, attributes[self.class.attribute_map[key]]))
        end # or else data not found in attributes(hash), not an issue as the data can be optional
      end

      self
    end

    # Deserializes the data based on type
    # @param string type Data type
    # @param string value Value to be deserialized
    # @return [Object] Deserialized data
    def _deserialize(type, value)
      case type.to_sym
      when :DateTime
        DateTime.parse(value)
      when :Date
        Date.parse(value)
      when :String
        value.to_s
      when :Integer
        value.to_i
      when :Float
        value.to_f
      when :BOOLEAN
        if value.to_s =~ /\A(true|t|yes|y|1)\z/i
          true
        else
          false
        end
      when :Object
        # generic object (usually a Hash), return directly
        value
      when /\AArray<(?<inner_type>.+)>\z/
        inner_type = Regexp.last_match[:inner_type]
        value.map { |v| _deserialize(inner_type, v) }
      when /\AHash<(?<k_type>.+?), (?<v_type>.+)>\z/
        k_type = Regexp.last_match[:k_type]
        v_type = Regexp.last_match[:v_type]
        {}.tap do |hash|
          value.each do |k, v|
            hash[_deserialize(k_type, k)] = _deserialize(v_type, v)
          end
        end
      else # model
        temp_model = SwaggerClient.const_get(type).new
        temp_model.build_from_hash(value)
      end
    end

    # Returns the string representation of the object
    # @return [String] String presentation of the object
    def to_s
      to_hash.to_s
    end

    # to_body is an alias to to_hash (backward compatibility)
    # @return [Hash] Returns the object in the form of hash
    def to_body
      to_hash
    end

    # Returns the object in the form of hash
    # @return [Hash] Returns the object in the form of hash
    def to_hash
      hash = {}
      self.class.attribute_map.each_pair do |attr, param|
        value = self.send(attr)
        next if value.nil?
        hash[param] = _to_hash(value)
      end
      hash
    end

    # Outputs non-array value in the form of hash
    # For object, use to_hash. Otherwise, just return the value
    # @param [Object] value Any valid value
    # @return [Hash] Returns the value in the form of hash
    def _to_hash(value)
      if value.is_a?(Array)
        value.compact.map { |v| _to_hash(v) }
      elsif value.is_a?(Hash)
        {}.tap do |hash|
          value.each { |k, v| hash[k] = _to_hash(v) }
        end
      elsif value.respond_to? :to_hash
        value.to_hash
      else
        value
      end
    end

  end
end

File: auto-generated/python/swagger_client/models/address.py
# coding: utf-8

"""
    BitMEX API

    ## REST API for the BitMEX Trading Platform  _If you are building automated tools, please subscribe to the_ _[BitMEX API RSS Feed](https://blog.bitmex.com/api_announcement/feed/) for changes. The feed will be updated_ _regularly and is the most reliable way to get downtime and update announcements._  [View Changelog](/app/apiChangelog)  -  #### Getting Started  Base URI: [https://www.bitmex.com/api/v1](/api/v1)  ##### Fetching Data  All REST endpoints are documented below. You can try out any query right from this interface.  Most table queries accept `count`, `start`, and `reverse` params. Set `reverse=true` to get rows newest-first.  Additional documentation regarding filters, timestamps, and authentication is available in [the main API documentation](/app/restAPI).  _All_ table data is available via the [Websocket](/app/wsAPI). We highly recommend using the socket if you want to have the quickest possible data without being subject to ratelimits.  ##### Return Types  By default, all data is returned as JSON. Send `?_format=csv` to get CSV data or `?_format=xml` to get XML data.  ##### Trade Data Queries  _This is only a small subset of what is available, to get you started._  Fill in the parameters and click the `Try it out!` button to try any of these queries.  - [Pricing Data](#!/Quote/Quote_get)  - [Trade Data](#!/Trade/Trade_get)  - [OrderBook Data](#!/OrderBook/OrderBook_getL2)  - [Settlement Data](#!/Settlement/Settlement_get)  - [Exchange Statistics](#!/Stats/Stats_history)  Every function of the BitMEX.com platform is exposed here and documented. Many more functions are available.  ##### Swagger Specification  [⇩ Download Swagger JSON](swagger.json)  -  ## All API Endpoints  Click to expand a section.   # noqa: E501

    OpenAPI spec version: 1.2.0
    Contact: support@bitmex.com
    Generated by: https://github.com/swagger-api/swagger-codegen.git
"""


import pprint
import re  # noqa: F401

import six

from swagger_client.configuration import Configuration


class Address(object):
    """NOTE: This class is auto generated by the swagger code generator program.

    Do not edit the class manually.
    """

    """
    Attributes:
      swagger_types (dict): The key is attribute name
                            and the value is attribute type.
      attribute_map (dict): The key is attribute name
                            and the value is json key in definition.
    """
    swagger_types = {
        'id': 'float',
        'currency': 'str',
        'created': 'datetime',
        'user_id': 'float',
        'address': 'str',
        'name': 'str',
        'note': 'str',
        'skip_confirm': 'bool',
        'skip_confirm_verified': 'bool',
        'skip2_fa': 'bool',
        'skip2_fa_verified': 'bool',
        'network': 'str',
        'memo': 'str'
    }

    attribute_map = {
        'id': 'id',
        'currency': 'currency',
        'created': 'created',
        'user_id': 'userId',
        'address': 'address',
        'name': 'name',
        'note': 'note',
        'skip_confirm': 'skipConfirm',
        'skip_confirm_verified': 'skipConfirmVerified',
        'skip2_fa': 'skip2FA',
        'skip2_fa_verified': 'skip2FAVerified',
        'network': 'network',
        'memo': 'memo'
    }

    def __init__(self, id=None, currency=None, created=None, user_id=None, address=None, name=None, note=None, skip_confirm=None, skip_confirm_verified=None, skip2_fa=None, skip2_fa_verified=None, network=None, memo=None, _configuration=None):  # noqa: E501
        """Address - a model defined in Swagger"""  # noqa: E501
        if _configuration is None:
            _configuration = Configuration()
        self._configuration = _configuration

        self._id = None
        self._currency = None
        self._created = None
        self._user_id = None
        self._address = None
        self._name = None
        self._note = None
        self._skip_confirm = None
        self._skip_confirm_verified = None
        self._skip2_fa = None
        self._skip2_fa_verified = None
        self._network = None
        self._memo = None
        self.discriminator = None

        if id is not None:
            self.id = id
        if currency is not None:
            self.currency = currency
        if created is not None:
            self.created = created
        if user_id is not None:
            self.user_id = user_id
        self.address = address
        self.name = name
        if note is not None:
            self.note = note
        if skip_confirm is not None:
            self.skip_confirm = skip_confirm
        if skip_confirm_verified is not None:
            self.skip_confirm_verified = skip_confirm_verified
        if skip2_fa is not None:
            self.skip2_fa = skip2_fa
        if skip2_fa_verified is not None:
            self.skip2_fa_verified = skip2_fa_verified
        self.network = network
        if memo is not None:
            self.memo = memo

    @property
    def id(self):
        """Gets the id of this Address.  # noqa: E501


        :return: The id of this Address.  # noqa: E501
        :rtype: float
        """
        return self._id

    @id.setter
    def id(self, id):
        """Sets the id of this Address.


        :param id: The id of this Address.  # noqa: E501
        :type: float
        """

        self._id = id

    @property
    def currency(self):
        """Gets the currency of this Address.  # noqa: E501


        :return: The currency of this Address.  # noqa: E501
        :rtype: str
        """
        return self._currency

    @currency.setter
    def currency(self, currency):
        """Sets the currency of this Address.


        :param currency: The currency of this Address.  # noqa: E501
        :type: str
        """

        self._currency = currency

    @property
    def created(self):
        """Gets the created of this Address.  # noqa: E501


        :return: The created of this Address.  # noqa: E501
        :rtype: datetime
        """
        return self._created

    @created.setter
    def created(self, created):
        """Sets the created of this Address.


        :param created: The created of this Address.  # noqa: E501
        :type: datetime
        """

        self._created = created

    @property
    def user_id(self):
        """Gets the user_id of this Address.  # noqa: E501


        :return: The user_id of this Address.  # noqa: E501
        :rtype: float
        """
        return self._user_id

    @user_id.setter
    def user_id(self, user_id):
        """Sets the user_id of this Address.


        :param user_id: The user_id of this Address.  # noqa: E501
        :type: float
        """

        self._user_id = user_id

    @property
    def address(self):
        """Gets the address of this Address.  # noqa: E501


        :return: The address of this Address.  # noqa: E501
        :rtype: str
        """
        return self._address

    @address.setter
    def address(self, address):
        """Sets the address of this Address.


        :param address: The address of this Address.  # noqa: E501
        :type: str
        """
        if self._configuration.client_side_validation and address is None:
            raise ValueError("Invalid value for `address`, must not be `None`")  # noqa: E501

        self._address = address

    @property
    def name(self):
        """Gets the name of this Address.  # noqa: E501


        :return: The name of this Address.  # noqa: E501
        :rtype: str
        """
        return self._name

    @name.setter
    def name(self, name):
        """Sets the name of this Address.


        :param name: The name of this Address.  # noqa: E501
        :type: str
        """
        if self._configuration.client_side_validation and name is None:
            raise ValueError("Invalid value for `name`, must not be `None`")  # noqa: E501

        self._name = name

    @property
    def note(self):
        """Gets the note of this Address.  # noqa: E501


        :return: The note of this Address.  # noqa: E501
        :rtype: str
        """
        return self._note

    @note.setter
    def note(self, note):
        """Sets the note of this Address.


        :param note: The note of this Address.  # noqa: E501
        :type: str
        """

        self._note = note

    @property
    def skip_confirm(self):
        """Gets the skip_confirm of this Address.  # noqa: E501


        :return: The skip_confirm of this Address.  # noqa: E501
        :rtype: bool
        """
        return self._skip_confirm

    @skip_confirm.setter
    def skip_confirm(self, skip_confirm):
        """Sets the skip_confirm of this Address.


        :param skip_confirm: The skip_confirm of this Address.  # noqa: E501
        :type: bool
        """

        self._skip_confirm = skip_confirm

    @property
    def skip_confirm_verified(self):
        """Gets the skip_confirm_verified of this Address.  # noqa: E501


        :return: The skip_confirm_verified of this Address.  # noqa: E501
        :rtype: bool
        """
        return self._skip_confirm_verified

    @skip_confirm_verified.setter
    def skip_confirm_verified(self, skip_confirm_verified):
        """Sets the skip_confirm_verified of this Address.


        :param skip_confirm_verified: The skip_confirm_verified of this Address.  # noqa: E501
        :type: bool
        """

        self._skip_confirm_verified = skip_confirm_verified

    @property
    def skip2_fa(self):
        """Gets the skip2_fa of this Address.  # noqa: E501


        :return: The skip2_fa of this Address.  # noqa: E501
        :rtype: bool
        """
        return self._skip2_fa

    @skip2_fa.setter
    def skip2_fa(self, skip2_fa):
        """Sets the skip2_fa of this Address.


        :param skip2_fa: The skip2_fa of this Address.  # noqa: E501
        :type: bool
        """

        self._skip2_fa = skip2_fa

    @property
    def skip2_fa_verified(self):
        """Gets the skip2_fa_verified of this Address.  # noqa: E501


        :return: The skip2_fa_verified of this Address.  # noqa: E501
        :rtype: bool
        """
        return self._skip2_fa_verified

    @skip2_fa_verified.setter
    def skip2_fa_verified(self, skip2_fa_verified):
        """Sets the skip2_fa_verified of this Address.


        :param skip2_fa_verified: The skip2_fa_verified of this Address.  # noqa: E501
        :type: bool
        """

        self._skip2_fa_verified = skip2_fa_verified

    @property
    def network(self):
        """Gets the network of this Address.  # noqa: E501


        :return: The network of this Address.  # noqa: E501
        :rtype: str
        """
        return self._network

    @network.setter
    def network(self, network):
        """Sets the network of this Address.


        :param network: The network of this Address.  # noqa: E501
        :type: str
        """
        if self._configuration.client_side_validation and network is None:
            raise ValueError("Invalid value for `network`, must not be `None`")  # noqa: E501

        self._network = network

    @property
    def memo(self):
        """Gets the memo of this Address.  # noqa: E501


        :return: The memo of this Address.  # noqa: E501
        :rtype: str
        """
        return self._memo

    @memo.setter
    def memo(self, memo):
        """Sets the memo of this Address.


        :param memo: The memo of this Address.  # noqa: E501
        :type: str
        """

        self._memo = memo

    def to_dict(self):
        """Returns the model properties as a dict"""
        result = {}

        for attr, _ in six.iteritems(self.swagger_types):
            value = getattr(self, attr)
            if isinstance(value, list):
                result[attr] = list(map(
                    lambda x: x.to_dict() if hasattr(x, "to_dict") else x,
                    value
                ))
            elif hasattr(value, "to_dict"):
                result[attr] = value.to_dict()
            elif isinstance(value, dict):
                result[attr] = dict(map(
                    lambda item: (item[0], item[1].to_dict())
                    if hasattr(item[1], "to_dict") else item,
                    value.items()
                ))
            else:
                result[attr] = value
        if issubclass(Address, dict):
            for key, value in self.items():
                result[key] = value

        return result

    def to_str(self):
        """Returns the string representation of the model"""
        return pprint.pformat(self.to_dict())

    def __repr__(self):
        """For `print` and `pprint`"""
        return self.to_str()

    def __eq__(self, other):
        """Returns true if both objects are equal"""
        if not isinstance(other, Address):
            return False

        return self.to_dict() == other.to_dict()

    def __ne__(self, other):
        """Returns true if both objects are not equal"""
        if not isinstance(other, Address):
            return True

        return self.to_dict() != other.to_dict()
